lib/humanized_id/humanizer.rb

```diff
@@ -0,0 +1,43 @@
+require 'facets'
+
+module HumanizedId
+  class Humanizer
+    def initialize(id:, length: nil, prefix: '')
+      @id               = id
+      @length           = length || id.to_s.length
+      @prefix           = prefix
+      @target_charset   = HumanizedId::CHARACTERSET
+    end
+
+    def generate_humanized_id
+      new_id = translate_id_to_charset
+      new_id = resize(id: new_id)
+      "#{@prefix}#{new_id}"
+    end
+
+    private
+
+    def translate_id_to_charset(id: @id, target_charset: @target_charset)
```

Comment:
Neat. So effectively this passes instance variables by value into local variables. But why is this necessary, why not just use the instance variables? `to_s` is not destructive, neither is picking a new char.